### PR TITLE
Teams branch

### PR DESCRIPTION
**Pull request by kevin-newland**
the pull request adds: 
teams_spec.rb: tests(initialize, Self.load_csv)
teams.rb: methods(initialize, self.load_csv)
fixture_teams.csv

### CODE REVIEW

File: spec/teams_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      expect(team1.team_id).to eq(1)
+      expect(team1.teamName).to eq("Atlanta United")
+    end
+  end
+
+  describe '#self.load_csv' do
+    it 'creates team from path and adds them to array through iteration' do
+      file_path = './spec/fixtures/fixture_teams.csv'
+      teams = Teams.load_csv(file_path)
+
```
Comment (DustinPeukert) [inline spec/teams_spec.rb:22]: you could add an assertion line here that also checks the correct number are being created, this would add just a bit more roundness to your testing.

**Comment by DustinPeukert** [comment]
I left a couple comments on changes that could be made, but you did a great job Kevin! I can see you've been improving and I'm proud of you. Keep up the great work!

File: lib/teams.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+              :teamName
+
+
+  def initialize(team_id, teamName)
+    @team_id = team_id
+    @teamName = teamName
+  end
+
+  def self.load_csv(file_path)
+    teams = []
```
Comment (DustinPeukert) [inline lib/teams.rb]: Instead of a teams array being returned here, I would instance create a class variable that stores all these instances. That way we can have another method that returns said class variable. Otherwise we could only ever access this data by creating it over and over again. Which could increase our runtimes.

Key points:
- class variable (likely named @@instances or @@teams)
- a method (.all) that returns this class variable

File: lib/teams.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    CSV.foreach('./spec/fixtures/fixture_teams.csv', headers: true, header_converters: :symbol) do |row|
+      team = Teams.new(row[:team_id].to_i, row[:teamName].to_s)
```
Comment (DustinPeukert) [inline lib/teams.rb]: row[:teamName] should be row[:teamname]
- you do not need the .to_s because everything is default a string

File: lib/teams.rb
Comment (qoryhanisagal) [inline lib/teams.rb:1]: **Overview:**
The Teams class is designed to handle individual team data, with attributes such as team_id and teamName. The class also includes a class method `load_csv` to create instances of Teams from a CSV file.

**Positives:**

- **Clear Structure:** The class has a clear structure, focusing on setting up individual team data with team_id and teamName attributes.
- **Data Loading:** The `load_csv` method successfully loads data from the specified CSV file and creates Teams instances, making it easy to generate data from external files.

**Suggestions:**

1.	**Consistent Naming for Attributes:** Consider using `team_name` instead of `teamName` to adhere to Ruby’s snake_case convention.
 
2.	**Error Handling for CSV Loading:** It may be helpful to add error handling within load_csv, especially if there’s potential for missing or incorrect data in the CSV file.

3.	**Use of file_path in load_csv:** Although file_path is passed as a parameter, the code currently references a hardcoded path ('./spec/fixtures/fixture_teams.csv'). Ensure file_path is used instead to make the method more flexible.

File: spec/fixtures/fixture_teams.csv
Comment (qoryhanisagal) [inline spec/fixtures/fixture_teams.csv:1]: **Overview:**
The fixture file is set up with realistic team data, mimicking actual CSV data for teams. This setup allows for consistent and repeatable tests.

**Positives:**

- **Realistic Test Data:** The data is structured to include various teams with attributes like team_id, franchiseId, teamName, etc., making it useful for testing purposes.
- **Representative Data for Testing:** Including multiple rows with diverse data enables testing under various scenarios.

**Suggestions:**

1.	**File Path Clarification:** Ensure that the path to this fixture file is consistently used across all tests to avoid discrepancies.

2.	**Data Completeness:** Since only team_id and teamName are currently used in Teams, you might consider updating Teams to use additional attributes in future expansions.

File: spec/teams_spec.rb
Comment (qoryhanisagal) [inline spec/teams_spec.rb:1]: **Overview:**
The test file `teams_spec.rb` verifies the functionality of the Teams class, specifically testing initialization and the `load_csv` method.

**Positives:**

- **Test Coverage:** The tests cover both object initialization and data loading from a CSV file, which are essential aspects of the Teams class.
- **Use of Fixture File:** The use of `fixture_teams.csv` allows for consistent test data, which helps ensure predictable test results.

**Suggestions:**

1.	**Test for Edge Cases in `load_csv`:** Add tests to check behavior when the CSV contains incomplete or malformed data. For example:
                - How does it handle missing or blank `teamName`?
2.	**Inline Comments in Tests:** Add inline comments explaining the purpose of each test block to improve readability and clarify test intentions.
3.	**Clarify Expected Outcomes in Test Descriptions:** Adjust test descriptions for clarity. For example, in it `'creates team from path and adds them to array through iteration'`, consider specifying expected behaviors like verifying `team_id` or `teamName` values.

**Review by qoryhanisagal — COMMENTED** [review]
**Summary:**
This pull request introduces the foundational setup for managing team data within the project. It includes the `Teams` class, a `fixture CSV` file for consistent test data, and a test suite to validate the `Teams` class’s core functionality.

**Positives:**

- **Adherence to SRP:** The Teams class focuses on handling team-related data, adhering to the Single Responsibility Principle.
- **Effective Testing:** The test file ensures the main features of Teams (initialization and CSV loading) function as expected.

**Suggestions:**

1.	**Standardize Attribute Naming:** Adjust the teamName attribute to team_name for Ruby convention consistency.
2.	**Edge Case Tests:** Enhance the test suite with cases for incomplete or erroneous data in load_csv.
3.	**Commenting and Documentation:** Adding inline comments and short descriptions in both the class file and the test file will improve readability and aid future maintenance.

**Review by qoryhanisagal — COMMENTED** [review]
**Summary:**
This pull request introduces the foundational setup for managing team data within the project. It includes the Teams class, a fixture CSV file for consistent test data, and a test suite to validate the Teams class’s core functionality.

**Positives:**

- **Adherence to SRP:** The Teams class focuses on handling team-related data, adhering to the Single Responsibility Principle.
- **Effective Testing:** The test file ensures the main features of Teams (initialization and CSV loading) function as expected.

**Suggestions:**

1.	**Standardize Attribute Naming:** Adjust the teamName attribute to team_name for Ruby convention consistency.
2.	**Edge Case Tests:** Enhance the test suite with cases for incomplete or erroneous data in load_csv.
3.	**Commenting and Documentation:** Adding inline comments and short descriptions in both the class file and the test file will improve readability and aid future maintenance.

**Review by KMPWeiland — COMMENTED** [review]
Add additional tests for the 'still has attributes' test just to make sure the data is being tested properly. Otherwise looks great!

**Review by KMPWeiland — COMMENTED** [review]
Nice test additions!

**Review by DustinPeukert — APPROVED** [review]
Looks great Kevin!

**Review by KMPWeiland — APPROVED** [review]
Nice work!

**Review by qoryhanisagal — APPROVED** [review]
**Overview:**
The Teams class is designed to handle individual team data, with attributes such as team_id and teamName. The class also includes a class method `load_csv` to create instances of Teams from a CSV file.

**Positives:**

- **Clear Structure:** The class has a clear structure, focusing on setting up individual team data with team_id and teamName attributes.
- **Data Loading:** The `load_csv` method successfully loads data from the specified CSV file and creates Teams instances, making it easy to generate data from external files.

**Suggestions:**

1.	**Consistent Naming for Attributes:** Consider using `team_name` instead of `teamName` to adhere to Ruby’s snake_case convention.
 
2.	**Error Handling for CSV Loading:** It may be helpful to add error handling within load_csv, especially if there’s potential for missing or incorrect data in the CSV file.

3.	**Use of file_path in load_csv:** Although file_path is passed as a parameter, the code currently references a hardcoded path ('./spec/fixtures/fixture_teams.csv'). Ensure file_path is used instead to make the method more flexible.
